packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/Translator.java

```diff
@@ -233,6 +235,34 @@ static HashMap<String, Object> fromBillingResult(BillingResult billingResult) {
     return info;
   }
 
+  static HashMap<String, Object> fromUserChoiceDetails(UserChoiceDetails userChoiceDetails) {
+    HashMap<String, Object> info = new HashMap<>();
+    info.put("externalTransactionToken", userChoiceDetails.getExternalTransactionToken());
+    info.put("originalExternalTransactionId", userChoiceDetails.getOriginalExternalTransactionId());
+    info.put("productsList", fromProductsList(userChoiceDetails.getProducts()));
+    return info;
+  }
+
+  static List<HashMap<String, Object>> fromProductsList(List<Product> productsList) {
+    if (productsList.isEmpty()) {
+      return Collections.emptyList();
+    }
+    ArrayList<HashMap<String, Object>> output = new ArrayList<>();
+    for (Product product : productsList) {
+      output.add(fromProduct(product));
+    }
+    return output;
+  }
+
+  static HashMap<String, Object> fromProduct(Product product) {
+    HashMap<String, Object> info = new HashMap<>();
+    info.put("productId", product.getId());
+    info.put("productOfferToken", product.getOfferToken());
```

Comment:
Why do these two have an extra "product" prefix relative to Dart (but the type below matches directly)?

Reply:
Mismatch, test added.